wolfcrypt/src/evp.c

```diff
@@ -6803,6 +6805,870 @@ void wolfSSL_EVP_PKEY_free(WOLFSSL_EVP_PKEY* key)
         }
     }
 }
+#if defined(OPENSSL_EXTRA)
+
+/* Indent writes white spaces of the number specified by "indents"
+ * to the BIO. The number of white spaces is limited from 0 to
+ * EVP_PKEY_PRINT_INDENT_MAX.
+ * returns the amount written to BIO.
+ */
+static int Indent(WOLFSSL_BIO* out, int indents )
+{
+    int i;
+    char space = ' ';
+
+    if (indents < 0)
+        indents = 0;
+    else if (indents > EVP_PKEY_PRINT_INDENT_MAX)
+        indents = EVP_PKEY_PRINT_INDENT_MAX;
+
+    if (out == NULL)
+        return 0;
+
+    for (i = indents; i; i--) {
+        if (wolfSSL_BIO_write(out, &space, 1) < 0)
+            break;
+    }
+    return indents -i;
+}
+/* DumpElement dump byte-data specified by "input" to the "out".
+ * Each line has leading white spaces( "indent" gives the number ) plus
+ * four spaces, then hex coded 15 byte data with separator ":" follow.   
+ * Each line looks like: 
+ * "    00:e6:ab: --- 9f:ef:"
+ * Returns 1 on success, 0 on failure.
+ */
+static int DumpElement(WOLFSSL_BIO* out, const byte* input,
+    int inlen, int indent)
+{
+#ifdef WOLFSSL_SMALL_STACK
+    byte*  buff = NULL;
+#else
+    byte   buff[EVP_PKEY_PRINT_LINE_WIDTH_MAX] = { 0 };
+#endif /* WOLFSSL_SMALL_STACK */
+
+    word32 in = 0;
+    word32 i;
+    int    idx;
+    int    wsz = 0;
+    word32 line;
+    word32 len;
+    word32 left;
+    const  byte* point;
+    word32 outSz;
+    byte   outHex[3];
+
+    if (!out || !input)
+        return WOLFSSL_FAILURE;
+
+    if (indent < 0)
+        indent = 0;
+    if (indent > EVP_PKEY_PRINT_INDENT_MAX)
+        indent = EVP_PKEY_PRINT_INDENT_MAX;
+
+
+    point = input;
+    len  = inlen;
+    line = len / 15;
+    left = len % 15;
+
+#ifdef WOLFSSL_SMALL_STACK
+    buff = (byte*)XMALLOC(EVP_PKEY_PRINT_LINE_WIDTH_MAX, NULL,
+        DYNAMIC_TYPE_TMP_BUFFER);
+    if (!buff)
+        return WOLFSSL_FAILURE;
+#endif
+
+    /* print pub element */
+
+    while (line) {
+        idx = 0;
+        Indent(out, indent);
+
+        for (i = 0; i < 15; i++) { /* 15 byte per line*/
+            outSz = sizeof(outHex);
+            if (Base16_Encode((const byte*)&point[in++], 1, outHex, &outSz ) != 0)
+                return WOLFSSL_FAILURE;
```

Comment:
Can you please update the return values in this function to account for the small stack build. The pointer 'buff' should be free'd with that build.

Reply:
Changed the logic to reach the free-of-buffer statement even if an error occurs.